mjolnir/input/read_local_interaction.hpp

```diff
@@ -122,6 +124,108 @@ read_contact_interaction(const std::string& kind, const toml::value& local)
     }
 }
 
+// This is reading contact part of read_directional_contact_interaction function.
+template<typename traitsT, typename ... PotentialTs>
+typename std::enable_if<
+    sizeof...(PotentialTs) == 2, std::unique_ptr<LocalInteractionBase<traitsT>>
+    >::type
+read_directional_contact_interaction(
+    const std::string& kind, const toml::value& local, std::vector<std::string>&&)
+{
+    MJOLNIR_GET_DEFAULT_LOGGER();
+    using real_type = typename traitsT::real_type;
+
+    real_type margin = 0.5; // default value
+    if (local.as_table().count("margin") == 1)
+    {
+        margin = toml::find<real_type>(local, "margin");
+    }
+
+    const auto contact_potential = toml::find<std::string>(local, "potentials", "contact");
+    if(contact_potential == "GoContact")
+    {
+        MJOLNIR_LOG_NOTICE("-- contact potential function is 10-12 Go contact.");
+        using contact_potentialT = GoContactPotential<real_type>;
+
+        return make_unique<
+            DirectionalContactInteraction<
+                traitsT, PotentialTs..., contact_potentialT
+                >
+            >(kind,
+              read_directional_contact_potentials<
+                  4, real_type, PotentialTs..., contact_potentialT
+              >(local), margin);
+    }
+    else if(contact_potential == "Gaussian")
+    {
+        MJOLNIR_LOG_NOTICE("-- contact potential function is Gaussian.");
+        using contact_potentialT = GaussianPotential<real_type>;
+
+        return make_unique<
+            DirectionalContactInteraction<
+                traitsT, PotentialTs..., contact_potentialT
+                >
+            >(kind,
+              read_directional_contact_potentials<
+                  4, real_type, PotentialTs..., contact_potentialT
+              >(local), margin);
+    }
+    else
+    {
+        throw_exception<std::runtime_error>(toml::format_error("[error] "
+            "mjolnir::read_directional_contact_interaction: invalid contact potential",
+            toml::find<toml::value>(local, "potentials", "contact"), "here", {
+            "expected value is one of the following.",
+            "- \"GoContact\": r^12 - r^10 type native contact potential",
+            "- \"Gaussian\" : well-known gaussian potential"
+            }));
+    }
+}
+
+// This is reading angle parts of read_directional_contact_interaction function.
+template<typename traitsT, typename ... PotentialTs>
+typename std::enable_if<
+    sizeof...(PotentialTs) < 2, std::unique_ptr<LocalInteractionBase<traitsT>>
+    >::type
+read_directional_contact_interaction(
+    const std::string& kind, const toml::value& local, std::vector<std::string>&& angle_potential_keys = {"angle2", "angle1"})
```

Comment:
I think it is good to pass it explicitly in the line 537. This argument is important, so I don't think it is a good way to make it "optional".